### PR TITLE
Add `value_range` utilities.

### PR DESCRIPTION
**Pull request by LukeWood**
This will help us implement the `value_range` parameter
for all of our KPLs.

We can normalize our inputs at the start of each `call` method, then assume [0, 255] range internally, then transform back to the users' dtype and value range at the end.

### CODE REVIEW

File: keras_cv/utils/preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    the input tensor is always in the range [0, 255].  This function
+    should be used at the end of the function body before returning
+    the input tensor to the user.
+
+    Args:
+        images: the set of images to transform to the value range
+        value_range: the value range to transform into
+
+    Returns:
+        a new Tensor with values in the range [0, 255]
```
Comment (qlzh727) [inline keras_cv/utils/preprocessing.py:61]: I think the return value should be in `value_range`, rather than [0, 255].

File: keras_cv/utils/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    """transforms input Tensor into value_range.
+
+    This function is intended to be used in preprocessing layers that
+    rely upon color values.  This allows us to assume internally that
+    the input tensor is always in the range [0, 255].  This function
+    should be used at the end of the function body before returning
+    the input tensor to the user.
+
+    Args:
+        images: the set of images to transform to the value range
```
Comment (qlzh727) [inline keras_cv/utils/preprocessing.py:57]: The images should have the default value range, which is [0, 255].

File: keras_cv/utils/preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        value_range
+    )
+    images = tf.math.minimum(images + 10, 255)
+    images = keras_cv.utils.transform_to_value_range(images, value_range)
+    ```
+    """
+    images = tf.cast(images, dtype=tf.float32)
+    min_value, max_value = _unwrap_value_range(value_range)
+    images = (images - min_value) / (max_value - min_value)
+    return images * 255.0
```
Comment (qlzh727) [inline keras_cv/utils/preprocessing.py:44]: Did we change the dtype of the output? we should make it clear in the docstring.

File: keras_cv/utils/preprocessing.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+def transform_to_standard_range(images, value_range):
```
Comment (qlzh727) [inline keras_cv/utils/preprocessing.py:17]: I think this interface is somehow confusing, since the input param `value _range` is not the standard_range, and the default standard range [0, 255] is hidden/implied in the implementation.

My original thought was to have it like, where user need to explicitly specify the from/to value range.

```
def convert_image_plx_value(images, from_range, to_range)
```

File: keras_cv/utils/preprocessing.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    """transforms the input Tensor to range [0, 255].
```
Comment (qlzh727) [inline keras_cv/utils/preprocessing.py:18]: Another random thought, since we always expect user to convert from and to its original image range (the convert function comes with pair), is it possible to do this in a context?

```
with utils.standard_image_value(images, original_value_range) as images:
   tf.image.ops(images)

# in the final block of the context, convert the image back to original value range.
tf.other_ops(image) # the image should be convert back to its original value range.


```

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: keras_cv/utils/preprocessing.py (hunk 5 above)
Comment (LukeWood) [reply keras_cv/utils/preprocessing.py:18]: Let me see if I can make this work.  Really smart idea.

Comment (LukeWood) [reply keras_cv/utils/preprocessing.py:18]: Just tested - not possible to do in a context unfortunately.

**Comment by LukeWood** [comment]
Update to rely on a single function.  This is a much more explicit user API - good point.

File: keras_cv/utils/preprocessing.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    ```python
+    original_range = [0, 1]
+    target_range = [0, 255]
+    images = keras_cv.utils.preprocessing.transform_value_range(
+        images,
+        original_range,
+        target_range
+    )
+    images = tf.math.minimum(images + 10, 255)
+    images = keras_cv.utils.transform_value_range(
```
[the remark this answers is not in the document]
Comment (LukeWood) [reply keras_cv/utils/preprocessing.py:44]: Fixed

**Comment by LukeWood** [comment]
@qlzh727 addressed comments.  Thanks!

File: keras_cv/utils/preprocessing.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+def transform_value_range(images, original_range, target_range, dtype=tf.float32):
+    """transforms values in input tensor from original_range to target_range.
+
+    This function is intended to be used in preprocessing layers that
+    rely upon color values.  This allows us to assume internally that
+    the input tensor is always in the range [0, 255].
+
+    Args:
+        images: the set of images to transform to the target range range.
+        original_range: the value range to transform from.
```
Comment (bhack) [inline keras_cv/utils/preprocessing.py:26]: Do we need to have these ranges free?  Are we not going to practically support just 3 value ranges as in tests?

Comment (LukeWood) [reply keras_cv/utils/preprocessing.py:26]: We actually do need them to be free.  `value_range` will likely be a user supplied value and we can't make assumptions about their dataset/formatting.

Comment (bhack) [reply keras_cv/utils/preprocessing.py:26]: Ok I don't know what other ranges could be interesting other then `-1,1`, `0,1` and `0,255`. 
But If we want to have any free range is responsibility of the caller to validates the inputs as we could theoretically have `nan` in:

`images = (images - original_min_value) / (original_max_value - original_min_value)`

**Review by qlzh727 — APPROVED** [review]
Thanks for the PR.
